docs/source/format/Columnar.rst

```diff
@@ -765,6 +765,65 @@ application.
 We discuss dictionary encoding as it relates to serialization further
 below.
 
+.. _run-length-encoded-layout:
+
+Run-Length-encoded Layout
+-------------------------
+
+Run-Length is a data representation that represents data as sequences of the
+same value, called runs. Each run is represented as a value, and an integer
+describing how often this value is repeated.
+
+Any array can be run-length-encoded. A run-length encoded array has a single
+buffer holding as many signed 32-bit integers, as there are runs. The actual
```

Comment:
This is actually not true if we use 32 bits. An array whose length is larger than a 32 bit integer cannot be represented